### PR TITLE
NCP5623c RGB LED driver I2C address auto detect

### PR DESCRIPTION
**Pull request by InspiredBrandon**
**Describe problem solved by this pull request**
mRobotics GPS receivers such as the mRo GPS u-Blox Neo-M8N IST8308 have an RGB LED that does not currently work with PX4. The RGB LED is driven by an NCP5634c which has an I2C address of 0x38. PWM 0 and PWM 2 are swapped compared to the existing driver.

**Describe your solution**
This solution sets the I2C address and PWM registers in the NCP5623c driver based on a parameter called MRO_GPS_RECEIVER

### CODE REVIEW

**Comment by dagar** [comment]
Looks good, but let's do it without the mRo specifics since it's just a slightly different (completely valid) use of the NCP5634c.

At startup we can probe the additional address.

**Comment by InspiredBrandon** [comment]
@dagar I removed the mRo specifics. Is there an implementation of probing for additional addresses that I can use as a reference?

**Comment by dagar** [comment]
> @dagar I removed the mRo specifics. Is there an implementation of probing for additional addresses that I can use as a reference?

In the driver you can either probe all the possible addresses, or pass -a on the command line when starting. https://github.com/PX4/PX4-Autopilot/blob/76dedfec5d1201dd5779271247213fe75efea5b9/ROMFS/px4fmu_common/init.d/rcS#L203

**Review by dagar — CHANGES_REQUESTED** [review]
This isn't necessarily wrong, but I think it's more detail than users will care about.

What if we cheat slightly and simply do it based on I2C address? Then at startup we can probe both. Would that effectively solve everything without burdening users?

**Comment by InspiredBrandon** [comment]
> This isn't necessarily wrong, but I think it's more detail than users will care about.
> 
> What if we cheat slightly and simply do it based on I2C address? Then at startup we can probe both. Would that effectively solve everything without burdening users?

@dagar I made both of the changes you requested.

**Comment by dagar** [comment]
> > This isn't necessarily wrong, but I think it's more detail than users will care about.
> > What if we cheat slightly and simply do it based on I2C address? Then at startup we can probe both. Would that effectively solve everything without burdening users?
> 
> @dagar I made both of the changes you requested.

Great, let's get this in.

File: src/drivers/lights/rgbled_ncp5623c/rgbled_ncp5623c.cpp [hunk 1]
```diff
@@ -136,9 +144,20 @@ RGBLED_NCP5623C::init()
 int
 RGBLED_NCP5623C::probe()
 {
+	int status = PX4_ERROR;
 	_retries = 4;
+	status = write(NCP5623_LED_CURRENT, NCP5623_LED_OFF);
+	if (status == PX4_ERROR) {
+		_retries = 4;
```
Comment (dagar) [inline src/drivers/lights/rgbled_ncp5623c/rgbled_ncp5623c.cpp:151]:
```suggestion
```

**Comment by dagar** [comment]
Good to go?

**Comment by InspiredBrandon** [comment]
> Good to go?

Thanks, I'm good to go.

**Review by dagar — APPROVED** [review]

**Comment by davids5** [comment]
@InspiredBrandon - There was a change to this driver on master, that may effect you.

The issue was the wiring of the PWM to color being hard coded in the driver based on address. 

See https://github.com/PX4/PX4-Autopilot/commit/376b1c9987ce78b657994d22ca1bc7b9a895cf25#diff-781353d78fc4c13509564b9700c0d850f5217043987a37debaec336850fc4f7aR111-R120 

You will need to add -o 321 to the command line for your use case.

**Pull request merged**